### PR TITLE
windows: remove extended path prefix \\?\

### PR DESCRIPTION
**Pull request by kenhys**
There is a case that GetModuleFileNameW returns path which contains
extended path prefix \\?\ as module filename. It causes a bug that
plugin can't be found correctly because of this kind of extended path
prefix.

TODO: Support long path correctly on Windows. (Drop MAX_PATH limitation)

GitHub:#958

Reported by @yagisumi Thanks!!!

### CODE REVIEW

**Comment by kenhys** [comment]
@komainu8 Could you review this PR?

File: lib/windows.c [hunk 1]
```diff
@@ -43,6 +43,12 @@ grn_windows_base_dir(void)
 
         absolute_dll_filename[absolute_dll_filename_size] = L'\0';
 
+        if (!wcsncmp(absolute_dll_filename, L"\\\\?\\", 4)) {
```
[the remark this answers is not in the document]
Comment (kenhys) [reply lib/windows.c]: なるほど！と思いました。

[the remark this answers is not in the document]
Comment (kenhys) [reply lib/windows.c]: 修正しました。

**Comment by kenhys** [comment]
@komainu8 I've fixed it!

**Comment by kenhys** [comment]
@komainu8 Thanks!

File: lib/grn_windows.h [hunk 2]
```diff
@@ -19,6 +19,7 @@
 #pragma once
 
 #include "grn.h"
+#include "wchar.h"
```
Comment (kou) [inline lib/grn_windows.h]: Why is this included in `windows.c`?

Comment (kou) [reply lib/grn_windows.h]: Can we move this to `windows.c`?

Comment (kenhys) [reply lib/grn_windows.h]: because `wmemmove` requires it.
https://docs.microsoft.com/ja-jp/cpp/c-runtime-library/reference/memmove-wmemmove?view=vs-2019

Comment (kenhys) [reply lib/grn_windows.h]: I'll fix it.

File: lib/plugin.c [hunk 3]
```diff
@@ -440,7 +440,7 @@ grn_plugin_open(grn_ctx *ctx, const char *filename)
   } else {
     const char *label;
     label = grn_dl_open_error_label();
-    SERR("%s", label);
+    SERR("%s: <%.*s>", label, filename_size, filename);
```
Comment (kou) [inline lib/plugin.c]: `(int)filename_size`.

This breaks Mroonga CI: https://travis-ci.org/mroonga/mroonga/jobs/538586519#L1680

Comment (kou) [reply lib/plugin.c]: BTW, is this safe?
Does `filename`'s encoding always equal to `grn_ctx->encoding`?
For example, if this is local encoding, we need to convert encoding by `grn_encoding_convert_from_locale()`.

Comment (kenhys) [reply lib/plugin.c]: Oh, I didn't noticed before that commit. :disappointed:
